payroll/views.py

```diff
@@ -47,6 +47,25 @@ def error(request, error_code):
 class EmployerView(DetailView, ChartHelperMixin):
     context_object_name = 'entity'
 
+    def get(self, request, *args, **kwargs):
+        # Django creates a different cache entry for every combination of path
+        # and URL parameters. To keep the number of cache entries low, redirect
+        # requests without a data_year parameter to the URL with a parameter
+        # for the latest year available. This will prevent duplicate entries
+        # for effectively the same page (an employer page without a data_year
+        # parameter, which would show the most recent year, and an employer
+        # page with a data_year parameter for the most recent year).
+        if not self.request.GET.get('data_year'):
+            self.object = self.get_object()
+            latest_year = self.data_years()[0]
+
+            return redirect(
+                '{0}?data_year={1}'.format(request.path, latest_year),
+                permanent=True
```

Comment:
Hm, should this actually be permanent? `latest_year` can change, but on a very slow timescale (< 1x year).

> When you use a 301, Google removes the old page from their index and most value (link equity) from that page is transferred to the new one.

> That being said, it’s important to note that anytime you move a page from one URL to another, it will take search engines some time to notice the change and see any potential impact/change in rankings.

> When used correctly, a 302 redirect will not hurt your SEO efforts.

> When you choose this type of redirect, the original page remains indexed in Google and no value (link equity) is transferred to the new URL because Google knows this is just temporary. Thus you’ll retain any rankings, traffic value, and authority that page might have.

Source: https://www.searchenginejournal.com/301-vs-302-redirects-seo/299843/#close

Given this, it seems like this should actually be a 301, not a 302. Does that seem right, @fgregg?

Reply:
if we are going to a strategy where are the urls will be parameterized, then I would recommend doing that only once, i.e. not redirecting to something based upon last_year which will change next year.

from an SEO perspective, the ideal thing is to not change the urls. let me know if you want some help figuring out a caching strategy that does the require a url change.